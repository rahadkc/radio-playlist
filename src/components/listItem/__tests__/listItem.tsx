import { cleanup, render } from '@testing-library/react'
import ListItem from '../index'

const props = {
  index: 1,
  item: { id: 1, name: 'Arman', value: 105 },
  activeIndex: 1,
  handleClick: () => {}
}

describe('<ListItem> component', () => {
  afterEach(cleanup)

  test('render ListItem component', () => {
    render(<ListItem {...props} />)
  })

  test('render item in the document', () => {
    const itemName = props?.item?.name
    const { queryByText } = render(<ListItem {...props} />)
    expect(queryByText(itemName)).toBeInTheDocument()
    expect(queryByText(itemName)).toMatchSnapshot()
  })
})
