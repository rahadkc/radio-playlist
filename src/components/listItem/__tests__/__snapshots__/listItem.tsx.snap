// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem> component render item in the document 1`] = `
<div
  class="name"
>
  Arman
</div>
`;
